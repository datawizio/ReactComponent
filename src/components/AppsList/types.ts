export interface IClient {
  client_id: number;
  name: string;
  is_active: boolean;
}

export interface IButtonClickParams {
  appId: number;
  url: string;
  allowed: boolean;
}

export type onAppClick = (clientId: number, params: IButtonClickParams) => void;

export interface CardAppProps {
  app_id: number;
  name: string;
  logo: string;
  dark_logo: string;
  host: string;
  path: string;
  description: string;
  allowed?: boolean;
  clients?: IClient[];
  onButtonClick?: onAppClick;
  buttonText?: string;
  showButton?: boolean;
}

export interface AppsListProps {
  apps: CardAppProps[];
  loading?: boolean;
  onSelect?: onAppClick;
}
