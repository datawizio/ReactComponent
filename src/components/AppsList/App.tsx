import React, { useState, useContext } from "react";

import { Card, Col } from "antd";
import Select from "../Select";
import Button from "../Button";

import ConfigContext from "../ConfigProvider/context";

import "./index.less";

export interface CardAppProps {
  app_id: number;
  name: string;
  logo: string;
  host: string;
  path: string;
  description: string;
  allowed?: boolean;
  clients?: { id: number; name: string }[];
  onButtonClick?: (
    clientId: number,
    { appId: number, url: string, allowed: boolean }
  ) => void;
  buttonText?: string;
  showButton?: boolean;
}

export const App: React.FC<CardAppProps> = ({
  app_id,
  name,
  logo,
  description,
  clients,
  host,
  path,
  allowed,
  showButton = true,
  onButtonClick
}) => {
  const { translate } = useContext(ConfigContext);
  const [client, setClient] = useState<number>();

  const handleChangeClient = (value: any) => {
    setClient(value);
  };

  const getClient = (client: number) => {
    if (client) return client;
    if (clients && clients.length > 0) return clients[0].id;
    return null;
  };

  const handleButtonClick = () => {
    const url = `${host ? host : ""}${path ? path : ""}`;
    onButtonClick(getClient(client), { appId: app_id, url, allowed });
  };

  return (
    <Col xs={24} sm={24} md={12} lg={8} xl={8} xxl={6}>
      <Card className="card-app">
        <div className="card-app-logo">
          {logo ? (
            <img src={logo} alt={name} />
          ) : (
            <div className="card-app-name">{name}</div>
          )}
        </div>
        <div className="card-app-description">{translate(description)}</div>
        {clients && clients.length > 1 && (
          <div className="card-app-clients">
            <Select
              placeholder={translate("SELECT_CLIENT")}
              showSearch
              optionFilterProp="label"
              onChange={handleChangeClient}
              value={client}
            >
              {clients.map(client => (
                <Select.Option key={client.id} value={client.id}>
                  {client.name}
                </Select.Option>
              ))}
            </Select>
          </div>
        )}

        {showButton && (
          <div className="card-app-actions">
            <Button type={"primary"} block onClick={handleButtonClick}>
              {translate(allowed ? "GO_OVER" : "LEARN_MORE")}
            </Button>
          </div>
        )}
      </Card>
    </Col>
  );
};
