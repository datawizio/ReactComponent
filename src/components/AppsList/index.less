@import "~antd/es/card/style/index";
@import "~antd/es/grid/style/index";
@import "../Select/index.less";
@import "../Button/index.less";
@import "../../styles/variables";

.row-apps {
  margin: -8px -8px 8px;
  & > .ant-col {
    padding: 8px;
  }
  .no-apps {
    padding: 8px;
  }
}
.card-app {
  height: 100%;
  display: flex;
  .ant-card-body {
    display: flex;
    flex: 1;
    flex-direction: column;
  }
  &-logo {
    text-align: center;
    height: 50px;
    img {
      height: 50px;
    }
    @media (max-width: @screen-xs-max) {
      height: auto;
      img {
        height: auto;
        max-width: 150px;
      }
    }
  }
  &-description {
    flex: 1;
    text-align: center;
    padding: @padding-md 0;
    display: flex;
    align-items: center;
    margin: 0 auto;
  }
  &-clients {
    padding-bottom: @padding-md;
  }
}
