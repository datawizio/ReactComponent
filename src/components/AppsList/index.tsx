import React, { useContext } from "react";
import ConfigContext from "../ConfigProvider/context";
import { Row } from "antd";
import { App } from "./App";
import { AppsLoader } from "./AppsLoader";
import "./index.less";

export interface IApp {
  app_id: number | string;
  name: string;
  logo: string;
  dark_logo: string;
  host: string;
  path: string;
  description: string;
  clients?: {
    id: number;
    name: string;
    is_active: boolean;
    client_id?: number;
  }[];
  allowed?: boolean;
}

export interface AppsListProps {
  apps: IApp[];
  loading?: boolean;
  onSelect?: (clientId: number, params: any) => void;
  disabledApps?: number[];
}

const AppsList: React.FC<AppsListProps> = ({
  apps,
  loading,
  onSelect,
  disabledApps
}) => {
  const { translate } = useContext(ConfigContext);

  const handleButtonClick = React.useCallback(
    (clientId: number, params: any) => {
      onSelect(clientId, params);
    },
    [onSelect]
  );

  return (
    <Row className="row-apps">
      {loading ? (
        <AppsLoader />
      ) : apps.length > 0 ? (
        apps.map(app => (
          <App
            key={app.app_id}
            {...app}
            disabledApps={disabledApps}
            onButtonClick={handleButtonClick}
          />
        ))
      ) : (
        <div className="no-apps">{translate("HAVENT_APPS")}</div>
      )}
    </Row>
  );
};

export default AppsList;
