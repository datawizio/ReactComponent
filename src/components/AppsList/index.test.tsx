import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import AppsList from "./index";

const mockProps = {
  apps: [
    {
      app_id: "1",
      name: "app",
      logo: "logoPath",
      host: "app.com",
      path: "/",
      description: "description",
      clients: [{ client_id: 1, name: "name" }]
    }
  ],
  onSelect: jest.fn()
};

const setUp = (props?) => mount(<AppsList {...props} />);

describe("AppsList component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn()
    }))
  });

  it("Render AppsList correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("Simmulate Button click", () => {
    const button = component.find("App").first();
    button.props().onButtonClick();
    expect(component.props().onSelect).toBeCalled();
  });

  it("Render AppsList loader", () => {
    expect(setUp({ loading: true })).toMatchSnapshot();
  });
});
