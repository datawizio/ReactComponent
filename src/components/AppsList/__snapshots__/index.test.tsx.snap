// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsList component Render AppsList correctly 1`] = `
<AppsList
  apps={
    Array [
      Object {
        "app_id": "1",
        "clients": Array [
          Object {
            "client_id": 1,
            "name": "name",
          },
        ],
        "description": "description",
        "host": "app.com",
        "logo": "logoPath",
        "name": "app",
        "path": "/",
      },
    ]
  }
  onSelect={[MockFunction]}
>
  <Row
    className="row-apps"
  >
    <div
      className="ant-row row-apps"
      style={
        Object {
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
        }
      }
    >
      <App
        app_id="1"
        clients={
          Array [
            Object {
              "client_id": 1,
              "name": "name",
            },
          ]
        }
        description="description"
        host="app.com"
        key="1"
        logo="logoPath"
        name="app"
        onButtonClick={[Function]}
        path="/"
      >
        <Col
          lg={8}
          md={12}
          sm={24}
          xl={8}
          xs={24}
          xxl={6}
        >
          <div
            className="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-8 ant-col-xl-8 ant-col-xxl-6"
            style={Object {}}
          >
            <Card
              className="card-app"
            >
              <div
                className="ant-card ant-card-bordered card-app"
              >
                <div
                  className="ant-card-body"
                  style={Object {}}
                >
                  <div
                    className="card-app-logo"
                  >
                    <img
                      alt="app"
                      src="logoPath"
                    />
                  </div>
                  <div
                    className="card-app-description"
                  >
                    description
                  </div>
                  <div
                    className="card-app-actions"
                  >
                    <Button
                      block={true}
                      border={true}
                      disabled={false}
                      onClick={[Function]}
                      type="primary"
                    >
                      <Button
                        block={true}
                        className="dw-btn"
                        disabled={false}
                        onClick={[Function]}
                        type="primary"
                      >
                        <Wave>
                          <button
                            className="ant-btn ant-btn-primary ant-btn-block dw-btn"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            <LoadingIcon
                              existIcon={false}
                              loading={false}
                              prefixCls="ant-btn"
                            >
                              <CSSMotion
                                motionName="ant-btn-loading-icon-motion"
                                onAppearActive={[Function]}
                                onAppearStart={[Function]}
                                onEnterActive={[Function]}
                                onEnterStart={[Function]}
                                onLeaveActive={[Function]}
                                onLeaveStart={[Function]}
                                removeOnLeave={true}
                                visible={false}
                              >
                                <DomWrapper />
                              </CSSMotion>
                            </LoadingIcon>
                            <span
                              key=".0"
                            >
                              LEARN_MORE
                            </span>
                          </button>
                        </Wave>
                      </Button>
                    </Button>
                  </div>
                </div>
              </div>
            </Card>
          </div>
        </Col>
      </App>
    </div>
  </Row>
</AppsList>
`;

exports[`AppsList component Render AppsList loader 1`] = `
<AppsList
  loading={true}
>
  <Row
    className="row-apps"
  >
    <div
      className="ant-row row-apps"
      style={
        Object {
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
        }
      }
    >
      <AppsLoader>
        <Col
          lg={8}
          md={12}
          sm={24}
          xl={8}
          xs={24}
          xxl={6}
        >
          <div
            className="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-8 ant-col-xl-8 ant-col-xxl-6"
            style={Object {}}
          >
            <Card
              className="card-app"
              loading={true}
            >
              <div
                className="ant-card ant-card-loading ant-card-bordered card-app"
              >
                <div
                  className="ant-card-body"
                  style={Object {}}
                >
                  <div
                    className="ant-card-loading-content"
                  >
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={22}
                        >
                          <div
                            className="ant-col ant-col-22"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={8}
                        >
                          <div
                            className="ant-col ant-col-8"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={15}
                        >
                          <div
                            className="ant-col ant-col-15"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={6}
                        >
                          <div
                            className="ant-col ant-col-6"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={18}
                        >
                          <div
                            className="ant-col ant-col-18"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={13}
                        >
                          <div
                            className="ant-col ant-col-13"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={9}
                        >
                          <div
                            className="ant-col ant-col-9"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={4}
                        >
                          <div
                            className="ant-col ant-col-4"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={3}
                        >
                          <div
                            className="ant-col ant-col-3"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={16}
                        >
                          <div
                            className="ant-col ant-col-16"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                  </div>
                </div>
              </div>
            </Card>
          </div>
        </Col>
        <Col
          lg={8}
          md={12}
          sm={24}
          xl={8}
          xs={24}
          xxl={6}
        >
          <div
            className="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-8 ant-col-xl-8 ant-col-xxl-6"
            style={Object {}}
          >
            <Card
              className="card-app"
              loading={true}
            >
              <div
                className="ant-card ant-card-loading ant-card-bordered card-app"
              >
                <div
                  className="ant-card-body"
                  style={Object {}}
                >
                  <div
                    className="ant-card-loading-content"
                  >
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={22}
                        >
                          <div
                            className="ant-col ant-col-22"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={8}
                        >
                          <div
                            className="ant-col ant-col-8"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={15}
                        >
                          <div
                            className="ant-col ant-col-15"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={6}
                        >
                          <div
                            className="ant-col ant-col-6"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={18}
                        >
                          <div
                            className="ant-col ant-col-18"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={13}
                        >
                          <div
                            className="ant-col ant-col-13"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={9}
                        >
                          <div
                            className="ant-col ant-col-9"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={4}
                        >
                          <div
                            className="ant-col ant-col-4"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={3}
                        >
                          <div
                            className="ant-col ant-col-3"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={16}
                        >
                          <div
                            className="ant-col ant-col-16"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                  </div>
                </div>
              </div>
            </Card>
          </div>
        </Col>
        <Col
          lg={8}
          md={12}
          sm={24}
          xl={8}
          xs={24}
          xxl={6}
        >
          <div
            className="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-8 ant-col-xl-8 ant-col-xxl-6"
            style={Object {}}
          >
            <Card
              className="card-app"
              loading={true}
            >
              <div
                className="ant-card ant-card-loading ant-card-bordered card-app"
              >
                <div
                  className="ant-card-body"
                  style={Object {}}
                >
                  <div
                    className="ant-card-loading-content"
                  >
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={22}
                        >
                          <div
                            className="ant-col ant-col-22"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={8}
                        >
                          <div
                            className="ant-col ant-col-8"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={15}
                        >
                          <div
                            className="ant-col ant-col-15"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={6}
                        >
                          <div
                            className="ant-col ant-col-6"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={18}
                        >
                          <div
                            className="ant-col ant-col-18"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={13}
                        >
                          <div
                            className="ant-col ant-col-13"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={9}
                        >
                          <div
                            className="ant-col ant-col-9"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      gutter={8}
                    >
                      <div
                        className="ant-row"
                        style={
                          Object {
                            "marginBottom": undefined,
                            "marginLeft": -4,
                            "marginRight": -4,
                            "marginTop": undefined,
                          }
                        }
                      >
                        <Col
                          span={4}
                        >
                          <div
                            className="ant-col ant-col-4"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={3}
                        >
                          <div
                            className="ant-col ant-col-3"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                        <Col
                          span={16}
                        >
                          <div
                            className="ant-col ant-col-16"
                            style={
                              Object {
                                "paddingLeft": 4,
                                "paddingRight": 4,
                              }
                            }
                          >
                            <div
                              className="ant-card-loading-block"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                  </div>
                </div>
              </div>
            </Card>
          </div>
        </Col>
      </AppsLoader>
    </div>
  </Row>
</AppsList>
`;
